packages/main/src/TableVirtualizer.ts

```diff
@@ -0,0 +1,299 @@
+
+class TableVirtualizer extends UI5Element implements ITableFeature {
+	/**
+	 * Defines the height of the rows in the table.
+	 *
+	 * **Note:** This property is mandatory for the virtualization to work properly.
+	 *
+	 * @default 45
+	 * @public
+	 */
+	@property({ type: Number })
+	rowHeight = 45;
+
+	/**
+	 * Defines the total count of rows in the table.
+	 *
+	 * **Note:** This property is mandatory for the virtualization to work properly.
+	 *
+	 * @default 100
+	 * @public
+	 */
+	@property({ type: Number })
+	rowCount = 100;
+
+	/**
+	 * Defines the count of extra rows to be rendered at the top and bottom of the table.
+	 *
+	 * **Note:** This property is experimental and may be changed or deleted in the future.
+	 *
+	 * @default 0
+	 * @public
+	 */
+	@property({ type: Number })
+	extraRows = 0;
+
+	readonly identifier = "TableVirtualizer";
+
+	_table?: Table;
+	_lastRowPosition: number = 0;
+	_firstRowPosition: number = 0;
+	_visibleRowCount: number = 0;
+	_tabBlockingState: TabBlocking = TabBlocking.None;
+	_onRowInvalidateBound: (invalidationInfo: InvalidationInfo) => void;
+	_onScrollBound: () => void;
+
+	constructor() {
+		super();
+		this._onScrollBound = throttle(this._onScroll.bind(this));
+		this._onRowInvalidateBound = this._onRowInvalidate.bind(this);
+	}
+
+	onTableActivate(table: Table): void {
+		this._table = table;
+		this._scrollContainer.addEventListener("scroll", this._onScrollBound, { passive: true });
+		this._onScroll();
+	}
+
+	onAfterRendering(): void {
+		this._table && this._table._invalidate++;
+	}
+
+	onTableAfterRendering(): void {
+		if (!this._table) {
+			return;
+		}
+
+		this._updateRowsHeight();
+		if (this._tabBlockingState & TabBlocking.Released) {
+			const tabBlockingRow = this._table.rows.at(this._tabBlockingState & TabBlocking.Next ? -1 : 0) as HTMLElement;
+			const tabForwardingElement = getTabbableElements(tabBlockingRow).at(this._tabBlockingState & TabBlocking.Next ? 0 : -1);
+			this._tabBlockingState = TabBlocking.None;
+			(tabForwardingElement || tabBlockingRow).focus();
+		}
+	}
+
+	onExitDOM(): void {
+		this._scrollContainer.removeEventListener("scroll", this._onScrollBound);
+		this._table = undefined;
+	}
+
+	/**
+	 * Resets the virtualizer to its initial state and triggers the `range-change` event.
+	 * @public
+	 */
+	reset(): void {
+		this._lastRowPosition = -1;
+		this._firstRowPosition = -1;
+		if (this._scrollContainer.scrollTop > 0) {
+			this._scrollContainer.scrollTop = 0;
+		} else {
+			this._onScroll();
+		}
+	}
+
+	get _scrollContainer() {
+		return this._table!._tableElement;
+	}
+
+	get _rowsContainer() {
+		return this._table!.shadowRoot!.getElementById("rows")!;
+	}
+
+	_onScroll(): void {
+		if (!this._table) {
+			return;
+		}
+
+		const headerRow = this._table.headerRow[0];
+		const headerHeight = headerRow.offsetHeight;
+		let scrollTop = this._scrollContainer.scrollTop;
+		let scrollableHeight = this._scrollContainer.clientHeight;
+		if (headerRow.sticky) {
```

Comment:
There seems to be an issue with the virtualizer, if the sticky property is added during runtime to the header row:

![virtualizer_sticky](https://github.com/user-attachments/assets/af7d063f-2866-4f70-9104-dd48e74ca60d)

I'm fine with ignoring this for now and opening a separate issue to handle this in another PR (just to finally get this feature in, as this case is very very unlikely to happen in an actual app)